src/main/groovy/com/palantir/gradle/shadowjar/ShadowJarPlugin.java

```diff
@@ -111,6 +112,13 @@ private void setupShadowJarToShadeTheCorrectDependencies(
             conf.setCanBeResolved(false);
         });
 
+        // Tell the consistent versions plugin not to look at the unshaded configuration for the
+        // checkUnusedConstraints task,  It forces resolution before the configuration below is complete which causes
+        // an error under gradle8.
+        VersionRecommendationsExtension recExt =
+                project.getExtensions().getByType(VersionRecommendationsExtension.class);
+        recExt.excludeConfigurations("unshaded");
```

Comment:
This direct java object manipulation also doesn't work when people are using plugin blocks due to different classloaders (quite common for GCV for some reason). Hence using groovy in `ShadowJarVersionLock#excludeConfigurationFromVersionsPropsInjection`.